src/VAC/vpaint_global.h

```diff
@@ -15,6 +15,14 @@
 #else
 #  define Q_VPAINT_EXPORT
 #endif
-
+enum struct ShapeType {
+    CURVE,
+    LINE,
+    CIRCLE,
+    TRIANGLE,
+    RECTANGLE,
+    POLYGON,
+    NONE,
```

Comment:
If we do it on faces we can remove this as well.

Reply:
I think NONE can be used to handle any error condition or when we don't know shapetype